### ISSUE TITLE
crash upon updating 

### ISSUE THREAD

**Issue by pat600130** [AstroTechies/astro_modloader#25]
so i just downloaded the update and upon running it the window appears gives me the spinning cursor and closes  
[modloader_log.txt](https://github.com/user-attachments/files/18118982/modloader_log.txt)
if that helps
note: im on win 11 if that helps

**Comment by SardinasA** [comment]
Faulting application name: astro_modloader.exe, version: 0.1.13.0, time stamp: 0x675adb53
Faulting module name: astro_modloader.exe, version: 0.1.13.0, time stamp: 0x675adb53
Exception code: 0xc0000409
Fault offset: 0x000000000063efb3
Faulting process id: 0x0x17BC
Faulting application start time: 0x0x1DB4E56FEFAA206
Faulting application path: astro_modloader.exe
Faulting module path: astro_modloader.exe
Report Id: 34d7bb0e-b177-48ee-9c93-33cae6a40b8d
Faulting package full name:

**Comment by thr3zims** [comment]
I installed the update today and had this happen as well. I even tried running it as admin, but it still didn't work (I'm on Windows 10).

**Comment by atenfyr** [comment]
Are any of you able to also provide logs from the mod loader?

**Comment by RedstoneWizard08** [comment]
> so i just downloaded the update and upon running it the window appears gives me the spinning cursor and closes [modloader_log.txt](https://github.com/user-attachments/files/18118982/modloader_log.txt) if that helps note: im on win 11 if that helps

@atenfyr looks like it's having trouble reaching the server and killing itself after a timeout; it could be something else though.

Can anyone run the modloader with the environment variable `RUST_LOG` set to `debug` and `RUST_BACKTRACE` set to `1`? Google how to do it if you don't know (it's hard to explain because I'm typing from my phone right now lol)

**Comment by j0rng3n** [comment]
Had similar if not the same issue as well (Windows 10).

Decided to tinker around with the source, debug sessions consistently crashed with one of the following exceptions:

```
Exception has occurred: W32/0xC000041D
Unhandled exception at 0x00007FFD51A313F0 (combase.dll) in astro_modloader.exe: 0xC000041D: An unhandled exception was encountered during a user callback.

combase.dll!00007ffd51a31410() (Unknown Source:0)
UIAutomationCore.dll!00007ffd1d005eaf() (Unknown Source:0)
UIAutomationCore.dll!00007ffd1d069036() (Unknown Source:0)
UIAutomationCore.dll!00007ffd1cfef841() (Unknown Source:0)
astro_modloader.exe!windows::Windows::Win32::UI::Accessibility::UiaReturnRawElementProvider<windows::Windows::Win32::Foundation::HWND,windows::Windows::Win32::Foundation::WPARAM,windows::Windows::Win32::Foundation::LPARAM,ref$<windows::Windows::Win32::UI::Accessibility::IRawElementProviderSimple>>(windows::Windows::Win32::Foundation::HWND hwnd, windows::Windows::Win32::Foundation::WPARAM wparam, windows::Windows::Win32::Foundation::LPARAM lparam, windows::Windows::Win32::UI::Accessibility::IRawElementProviderSimple * el) Line 1370 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\windows-0.42.0\src\Windows\Win32\UI\Accessibility\mod.rs:1370)
astro_modloader.exe!accesskit_windows::adapter::impl$1::from(accesskit_windows::adapter::WmGetObjectResult this) Line 217 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\accesskit_windows-0.12.0\src\adapter.rs:217)
astro_modloader.exe!core::convert::impl$3::into<accesskit_windows::adapter::WmGetObjectResult,windows::Windows::Win32::Foundation::LRESULT>(accesskit_windows::adapter::WmGetObjectResult self, core::panic::location::Location *) Line 760 (f:\Dev\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\convert\mod.rs:760)
astro_modloader.exe!accesskit_windows::subclass::wnd_proc(windows::Windows::Win32::Foundation::HWND window, unsigned int message, windows::Windows::Win32::Foundation::WPARAM wparam, windows::Windows::Win32::Foundation::LPARAM lparam) Line 35 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\accesskit_windows-0.12.0\src\subclass.rs:35)
user32.dll!00007ffd51f1ef5c() (Unknown Source:0)
user32.dll!00007ffd51f1e8cc() (Unknown Source:0)
user32.dll!00007ffd51f310c3() (Unknown Source:0)
ntdll.dll!00007ffd536f1374() (Unknown Source:0)
win32u.dll!00007ffd50f01124() (Unknown Source:0)
user32.dll!00007ffd51f31044() (Unknown Source:0)
oleacc.dll!00007ffd3a9564d7() (Unknown Source:0)
oleacc.dll!00007ffd3a952b97() (Unknown Source:0)
oleacc.dll!00007ffd3a98dee4() (Unknown Source:0)
oleacc.dll!00007ffd3a98dbf6() (Unknown Source:0)
msctf.dll!00007ffd52b5ea4c() (Unknown Source:0)
msctf.dll!00007ffd52b5e779() (Unknown Source:0)
```

```
Exception has occurred: W32/0xC0000374
Unhandled exception at 0x00007FFD5374F3C9 (ntdll.dll) in astro_modloader.exe: 0xC0000374: A heap has been corrupted (parameters: 0x00007FFD537B97F0).

ntdll.dll!00007ffd5374f3c9() (Unknown Source:0)
ntdll.dll!00007ffd5374f393() (Unknown Source:0)
ntdll.dll!00007ffd53758112() (Unknown Source:0)
ntdll.dll!00007ffd537583fa() (Unknown Source:0)
ntdll.dll!00007ffd5375e081() (Unknown Source:0)
ntdll.dll!00007ffd53675bf0() (Unknown Source:0)
ntdll.dll!00007ffd536747b1() (Unknown Source:0)
astro_modloader.exe!alloc::alloc::impl$1::deallocate(alloc::alloc::Global * self, core::ptr::non_null::NonNull<u8> ptr, core::alloc::layout::Layout) Line 266 (f:\Dev\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\alloc.rs:266)
astro_modloader.exe!alloc::boxed::impl$8::drop<accesskit_windows::node::PlatformNode_Impl,alloc::alloc::Global>(accesskit_windows::node::PlatformNode_Impl * * self) Line 1678 (f:\Dev\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1678)
astro_modloader.exe!core::ptr::drop_in_place<alloc::boxed::Box<accesskit_windows::node::PlatformNode_Impl,alloc::alloc::Global>>(accesskit_windows::node::PlatformNode_Impl * *) Line 574 (f:\Dev\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ptr\mod.rs:574)
astro_modloader.exe!accesskit_windows::node::impl$6::Release(accesskit_windows::node::PlatformNode_Impl * self) Line 505 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\accesskit_windows-0.12.0\src\node.rs:505)
astro_modloader.exe!windows::core::unknown::impl$7::new::Release<accesskit_windows::node::PlatformNode_Impl,-2>(core::ffi::c_void * this) Line 99 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\windows-0.42.0\src\core\unknown.rs:99)
astro_modloader.exe!windows::core::unknown::impl$3::drop(windows::core::unknown::IUnknown * self) Line 41 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\windows-0.42.0\src\core\unknown.rs:41)
astro_modloader.exe!core::ptr::drop_in_place<windows::core::unknown::IUnknown>(windows::core::unknown::IUnknown *) Line 574 (f:\Dev\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ptr\mod.rs:574)
astro_modloader.exe!core::ptr::drop_in_place<windows::Windows::Win32::UI::Accessibility::IRawElementProviderSimple>(windows::Windows::Win32::UI::Accessibility::IRawElementProviderSimple *) Line 574 (f:\Dev\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ptr\mod.rs:574)
astro_modloader.exe!accesskit_windows::adapter::QueuedEvents::raise(accesskit_windows::adapter::QueuedEvents self) Line 241 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\accesskit_windows-0.12.0\src\adapter.rs:241)
astro_modloader.exe!accesskit_winit::platform_impl::platform::Adapter::update_if_active<egui_winit::impl$0::handle_platform_output::closure_env$0>(egui_winit::impl$0::handle_platform_output::closure_env$0 self) Line 32 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\accesskit_winit-0.10.0\src\platform_impl\windows.rs:32)
astro_modloader.exe!accesskit_winit::Adapter::update_if_active<egui_winit::impl$0::handle_platform_output::closure_env$0>(egui_winit::impl$0::handle_platform_output::closure_env$0 self) Line 133 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\accesskit_winit-0.10.0\src\lib.rs:133)
astro_modloader.exe!egui_winit::State::handle_platform_output(winit::window::Window * self, egui::context::Context * window, egui::data::output::PlatformOutput egui_ctx) Line 652 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\egui-winit-0.21.1\src\lib.rs:652)
astro_modloader.exe!eframe::native::epi_integration::EpiIntegration::handle_platform_output(winit::window::Window * self, egui::data::output::PlatformOutput window) Line 503 (f:\Dev\.cargo\registry\src\index.crates.io-6f17d22bba15001f\eframe-0.21.3\src\native\epi_integration.rs:503)
```

Accesskit is likely to blame. Updating eframe/egui to 0.29.1 and egui_commonmark to 0.18.0 fixes the issue for me, but also alters the UI a bit.
Other versions might work as well, just tested with these ones.

**Comment by RedstoneWizard08** [comment]
Well, that's certainly a long error 🤣 
Yeah the dependencies should probably be updated, I might try my hand at it

**Comment by atenfyr** [comment]
(Seemingly) fixed in Release v0.1.14
